.teamcity/_self/lib/customBuildType/E2EBuildType.kt

```diff
@@ -0,0 +1,170 @@
+		val buildParams = buildParams
+		val buildFeatures = buildFeatures
+		val buildTriggers = buildTriggers
+		val buildDependencies = buildDependencies
+		val params = params
+
+		id( buildId )
+		uuid = buildUuid
+		name = buildName
+		description = buildDescription
+		maxRunningBuilds = concurrentBuilds
+
+		artifactRules = """
+			logs.tgz => logs.tgz
+			screenshots => screenshots
+			trace => trace
+		""".trimIndent()
+
+		vcs {
+			root(Settings.WpCalypso)
+			cleanCheckout = true
+		}
+
+		params {
+			param("env.NODE_CONFIG_ENV", "test")
+			param("env.PLAYWRIGHT_BROWSERS_PATH", "0")
+			param("env.TEAMCITY_VERSION", "2021")
+			param("env.HEADLESS", "false")
+			param("env.LOCALE", "en")
+			param("env.DEBUG", "pw:api")
+			buildParams()
+		}
+
+		steps {
+			prepareEnvironment()
+
+			bashNodeScript {
+				name = "Run tests"
+				scriptContent = """
+					# Configure bash shell.
+					shopt -s globstar
+					set -x
+
+					# For Calypso E2E build configurations, the URL environment variable
+					# is computed and exported by a script.
+					$getCalypsoLiveURL
+
+					# Enter testing directory.
+					cd test/e2e
+					mkdir temp
+
+					# Decrypt config
+					openssl aes-256-cbc -md sha1 -d -in ./config/encrypted.enc -out ./config/local-test.json -k "%CONFIG_E2E_ENCRYPTION_KEY%"
+
+					# As noted above, Calypso E2E build configurations export the URL
+					# environment variable within this step - therefore the export
+					# for NODE_CONFIG variable has to be done here instead of
+					# within the Kotlin DSL.
+					export NODE_CONFIG="{\"calypsoBaseURL\":\"${'$'}{URL%/}\"}"
+
+					# Run suite.
+					xvfb-run yarn jest --reporters=jest-teamcity --reporters=default --maxWorkers=%E2E_WORKERS% --group=$testGroup
+				"""
+				dockerImage = "%docker_image_e2e%"
+			}
+
+			collectResults()
+		}
+
+		features {
+			perfmon {
+			}
+			commitStatusPublisher {
+				vcsRootExtId = "${Settings.WpCalypso.id}"
+				publisher = github {
+					githubUrl = "https://api.github.com"
+					authType = personalToken {
+						token = "credentialsJSON:57e22787-e451-48ed-9fea-b9bf30775b36"
+					}
+				}
+			}
+			buildFeatures()
+		}
+
+		// By default, no triggers are defined for this template class.
+		triggers {buildTriggers()}
+
+		dependencies.buildDependencies()
+
+		failureConditions {
+			executionTimeoutMin = 20
+			// Don't fail if the runner exists with a non zero code. This allows a build to pass if the failed tests have been muted previously.
+			nonZeroExitCode = false
+
+			// Fail if the number of passing tests is 50% or less than the last build. This will catch the case where the test runner crashes and no tests are run.
+			failOnMetricChange {
+				metric = BuildFailureOnMetric.MetricType.PASSED_TEST_COUNT
+				threshold = 50
+				units = BuildFailureOnMetric.MetricUnit.PERCENTS
+				comparison = BuildFailureOnMetric.MetricComparison.LESS
+				compareTo = build {
+					buildRule = lastSuccessful()
+				}
+			}
+		}
```

Comment:
Some builds previously only had the `executionTimeoutMin` and `nonZeroExitCode` failure conditions, but in a similar story as above I've chosen to include `failOnMetricChange` as another default failure condition.